### ISSUE TITLE
check_ntp_time calls `write` on a UDP socket without a successful call to `connect`

### ISSUE THREAD

**Issue by thanatos** [nagios-plugins/nagios-plugins#171]
There's some bad handling of socket state in `check_ntp_time` somewhere (everywhere?); for example, we see:

```
write(7, "\343\0\4\372\0\1\0\0\0\1\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 48) = -1 EDESTADDRREQ (Destination address required)
```

Here, `write` is failing b/c the socket is not connected, because the `connect` call fails:

```
connect(8, {sa_family=AF_INET6, sin6_port=htons(123), inet_pton(AF_INET6, "2001:67c:1560:8003::c7", &sin6_addr), sin6_flowinfo=0, sin6_scope_id=0}, 28) = -1 ENETUNREACH (Network is unreachable)
```

This call fails b/c our setup has no IPv6 support, because AWS has yet to get their act together. Unfortunately, this is going to be a disgustingly common setup, and it appears to also make `check_ntp_time` take longer than it needs to (because it thinks it needs to poll the socket, b/c it thinks it is waiting for a response to a request it never sent).

The code [branches on error](https://github.com/nagios-plugins/nagios-plugins/blob/c96812728a9aacd77402cff603c5c64cbba606c1/plugins/check_ntp_time.c#L348-L353), but doesn't actually end up _handling_ the error, aside from a log (that I don't actually ever see, presumably because `DBG` requires a debug build? if this is the case, I'd _highly_ recommend exceptional should-probably-not-happen errors like connection failures get logged normally, or at least logged if I pass `-v`…)

**Comment by jfrickson** [comment]
Looking into this.

**Comment by jfrickson** [comment]
Fixed in branch `maint` via commit https://github.com/nagios-plugins/nagios-plugins/commit/7832f95eb625cba93841c942ead4e172526b0d2f.

Will be released in version 2.2.2
